db.go

```diff
@@ -422,14 +422,20 @@ func (db *DB) Close() (err error) {
 		}
 	}
 
-	// Ensure replicas perform a final sync and stop replicating.
+	// Ensure replicas stop replicating and perform a final sync.
 	for _, r := range db.Replicas {
+		// Stop normal background sync.
+		r.Stop()
+
+		// Force one final sync if DB is open.
 		if db.db != nil {
 			if e := r.Sync(ctx); e != nil && err == nil {
 				err = e
 			}
 		}
-		if e := r.Stop(true); e != nil && err == nil {
+
+		// Close out replica.
+		if e := r.Close(); e != nil && err == nil {
 			err = e
 		}
 	}
```

Comment:
Previously, this tried to sync before stopping the replica but that could cause the replica to sync twice at the same time. Now it stops, syncs, & closes.